src/utils/api.ts

```diff
@@ -0,0 +1,30 @@
+// Copyright 2018 - 2021 The Alephium Authors
```

Comment:
@mvaivre this file and the next one are a good example of duplicated code between our projects.

I copied them from the alephium-wallet project. Since I started using alephium-js in this project, our error handling needs to adapt to the responses of the alephium-js API classes. This is how the wallet handles it and I think this is how the explorer should handle it as well.

The question is: How do we handle shared logic between our projects? Is it too early to start thinking of shared codebases? Maybe we could add these 2 files in the alephium-js?

Tbh, it makes me feel a bit dirty copying the same files amongst projects, but maybe it's the best approach for the moment. Happy to discuss.

Reply:
Note to myself: I'll probably need to update the code to reflect this bug fix: https://github.com/alephium/alephium-wallet/pull/94

Reply:
Everything started in the explorer before going into the wallet. As you saw, some things changed along the way (error format etc.), everything was constantly moving so yeah - it was too soon to put everything in a shared codebase.

Now might be the time to go back to centralising. 
Error management and amount display could be a good start.

It would be good to discuss how with structure `alephium-js` in future iterations though. For now, it was essentially focused on api methods (flat hierarchy). Now we're gonna start adding UI helpers. Distinction should be made clear.

Reply:
Yeah, if we want to "sell" the alephium-js project as the SDK for building smart contracts on Alephium, then UI helpers definitely have no place there.

What we essentially see emerging is common functionality (as you correctly said, error formating, amount display, etc) amongst our React apps (explorer, wallet). We will build more React apps, including the dapps.

Maybe adding a couple UI-related shared functions to alephium-js is not that bad of an idea to start with. The more we find common functionality, the more ideas will be generated as to how to handle separation of concerns.

Reply:
Constants should also be in alephium-js. I'm about to add some in the wallet (minGas etc.). I don't want to add them to the SDK right away, as we need to discuss this together first. 

But to summarise quickly, I see:

- API methods
- Formatting helpers (numbers, amounts, errors etc.)
- Constants

.. being thrown in alephium-js. 

I'll create a task there to continue the discussion 🚀